source/slang/slang-parser.cpp

```diff
@@ -4364,6 +4382,46 @@ namespace Slang
         return attrDecl;
     }
 
+    static void addSpecialGLSLModifiersBasedOnType(
+        Parser* parser,
+        Decl* decl,
+        Modifiers* modifiers)
+    {
+        // we intentionally have findModifier twice since GLSLOffsetLayoutAttribute is rare
+        // better to early leave
+        if (auto bindingMod = modifiers->findModifier<GLSLBindingAttribute>())
+        {
+            if (auto varDeclBase = as<VarDeclBase>(decl))
+            {
+                if (auto declRefExpr = as<DeclRefExpr>(varDeclBase->type.exp))
+                {
```

Comment:
I don't see why your comment is needed here.
Calling an expensive function later is a common practice in programming.

And I like to suggest the following code for less indentation.
```
auto varDeclBase = as<VarDeclBase>(decl));
if (!varDeclBase) return;

auto declRefExpr = as<DeclRefExpr>(varDeclBase->type.exp);
if (!declRefExpr) return;

auto name = declRefExpr->name;
if (!name) return;
if (!name->text.equals("atomic_uint")) return;

auto bindingMod = modifiers->findModifier<GLSLBindingAttribute>();
if (!bindingMod) return;
```